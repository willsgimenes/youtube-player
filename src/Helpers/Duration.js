export const prettyDuration = time => {
  const string = time.match(/(\d+)(?=[MHS])/ig) || []

  return string.map(function (item) {
    if (item.length < 2) return '0' + item
    return item
  }).join(':')
}

export const converToSeconds = time => {
  const string = time.match(/(\d+)(?=[MHS])/ig) || []

  const parsed = string.map(function (item) {
    if (item.length < 2) return '0' + item
    return item
  }).join(':')

  return parsed.split(':').reduce((acc, time) => (60 * acc) + +time)
}
