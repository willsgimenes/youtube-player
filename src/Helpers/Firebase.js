import Axios from 'axios'

const BASE_URL = 'https://f-player.herokuapp.com/'

export const fetchIndex = async () => {
  return Axios.get(`${BASE_URL}songs/index`)
}

export const fetchSongs = async () => {
  return Axios.get(`${BASE_URL}songs/list`)
}

export const postList = async song => {
  return Axios.post(`${BASE_URL}songs`, song)
}
