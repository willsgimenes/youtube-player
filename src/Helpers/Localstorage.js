/* global localStorage */

export const loadState = (state) => {
  try {
    const serializedState = localStorage.getItem(state)

    if (serializedState === null) {
      return undefined
    }
    return JSON.parse(serializedState)
  } catch (err) {
    return {}
  }
}

export const loadStateNotSerialized = (state) => {
  const serializedState = localStorage.getItem(state)
  if (serializedState === null) {
    return undefined
  }
  return serializedState
}

export const saveState = (state, value) => {
  let serializedState = value

  if (typeof serializedState === 'string') {
    localStorage.setItem(state, serializedState)
  } else {
    serializedState = JSON.stringify(value)
    localStorage.setItem(state, serializedState)
  }
}

export const removeItem = (key) => {
  localStorage.removeItem(key)
}
