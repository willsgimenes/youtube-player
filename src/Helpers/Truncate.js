export default function (string, length) {
  var truncatedString = string

  if (string.length >= length) { truncatedString = string.substring(0, length - 3) + '...' }

  return truncatedString
}
