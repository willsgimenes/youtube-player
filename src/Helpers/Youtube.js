import Axios from 'axios'

const fetchVideo = async term => {
  return Axios.get('https://f-player.herokuapp.com/', {
    params: {
      term: term
    }
  })
}

const getVideoByTitle = (term) => {
  return new Promise((resolve, reject) => {
    fetchVideo(term)
      .then(({ data }) => resolve(data))
      .catch(error => reject(error))
  })
}

export { getVideoByTitle, fetchVideo }
