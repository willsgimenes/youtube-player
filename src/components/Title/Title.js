import React from 'react'
import { loadState } from '../../Helpers/Localstorage'

const Title = (props) => {
  const id = loadState('StreamItems-' + props.Id) || null

  if (id !== null) {
    return (
      <p className='truncate'>{id.title}</p>
    )
  } else {
    return (
      <p>no title</p>
    )
  }
}

export default Title
