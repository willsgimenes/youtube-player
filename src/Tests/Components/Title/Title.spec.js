/* global describe */
/* global beforeEach */
/* global it */
/* global expect */

import React from 'react'
import Enzyme, { shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'

import Title from '../../../components/Title/Title'

import toJson from 'enzyme-to-json'
import 'jest-styled-components'

import { saveState } from '../../../Helpers/Localstorage'

function storageMock () {
  var storage = {}

  return {
    setItem: function (key, value) {
      storage[key] = value || ''
    },
    getItem: function (key) {
      return key in storage ? storage[key] : null
    },
    removeItem: function (key) {
      delete storage[key]
    },
    get length () {
      return Object.keys(storage).length
    },
    key: function (i) {
      var keys = Object.keys(storage)
      return keys[i] || null
    }
  }
}

window.localStorage = storageMock()

saveState(
  'StreamItems-ac0d862c-d702-4b13-7150-5f982a590f98',
  {
    id: 'ac0d862c-d702-4b13-7150-5f982a590f98',
    song: {
      url: 'foo',
      duration: 133
    }
  }
)

saveState(
  'StreamItems-8e5cde91-5170-f6fd-b210-eeda77e5a00e',
  {
    id: '8e5cde91-5170-f6fd-b210-eeda77e5a00e',
    song: {
      url: 'bar',
      duration: 133
    }
  }
)

Enzyme.configure({ adapter: new Adapter() })

describe('<Title />', () => {
  let wrapper

  beforeEach(() => {
    wrapper = shallow(<Title Id='ac0d862c-d702-4b13-7150-5f982a590f98' />)
  })

  it('should render properly', () => {
    expect(wrapper.find('p')).toHaveLength(1)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('should not render title', () => {
    wrapper = shallow(<Title Id='ac0d862c-d702-4b13-7150-5f982a590f91' />)
    expect(wrapper.find('p')).toHaveLength(1)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
