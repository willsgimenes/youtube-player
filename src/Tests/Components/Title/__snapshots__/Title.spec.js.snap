// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Title /> should not render title 1`] = `
<p>
  no title
</p>
`;

exports[`<Title /> should render properly 1`] = `
<p
  className="truncate"
/>
`;
